src/i18n.mjs

```diff
@@ -74,7 +74,16 @@ class I18n {
   }
 
   setLocale (locale) {
-    this.currentLocale = locale
+
+    // Allow validation and re-writing of the local string
+    const candidate = this.whenChangingLocale(locale)
+
+    // Do not update locale if new value is null or undefined
+    if (candidate !== undefined && candidate !== null) {
+      const newLocale = String(candidate)
```

Comment:
Not sure why you need to cast it to a `String`, it should be a string value. 

With Ficus, I've tried to assume that the library will be used correctly for it's intended purpose. Therefore, I do not add lots of validation, casting or type conversion in order to keep the code as light as possible. If someone implements the library incorrectly, errors will naturally surface and it will be clear from testing that it is not behaving as intended.

Reply:
You may agree or disagree with this viewpoint. I have learnt over many years and experienced many bloated libraries that over compensate for poor implementations by adding too much validation and type conversion code.

Happy to share more on some of the decisions employed in creating Ficus to give context.

Reply:
I totally agree with:
> If someone implements the library incorrectly, errors will naturally surface and it will be clear from testing that it is not behaving as intended. [...] I have learnt over many years and experienced many bloated libraries that over compensate for poor implementations by adding too much validation and type conversion code.

This was partly the point of my questions above about `string | null | undefined` vs `any` type, as well as [this question](https://github.com/ficusjs/ficusjs-i18n/issues/55#issuecomment-1652182244). I'm happy to update the code if assuming that the argument is of `string` type is more in line with FicusJS' philosophy. Plus, this keeps disturbance of the current API to a minimum.

Reply:
Btw, I'm very interested in learning about the design decisions employed in creating FicusJS. That will certainly help me better orient my coding right from the start.

Reply:
If the `null`/`undefined` check passes, you can assume the value is a string and use it asis. That should be sufficient.